python/cuvs/cuvs/test/test_brute_force.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) 2022-2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+import pytest
+from pylibraft.common import device_ndarray
+from scipy.spatial.distance import cdist
+
+from cuvs.neighbors import brute_force
+
+
+@pytest.mark.parametrize("n_index_rows", [32, 100])
+@pytest.mark.parametrize("n_query_rows", [32, 100])
+@pytest.mark.parametrize("n_cols", [40, 100])
```

Comment:
Same question as why these particular values. Perhaps we're overparametrizing a bit, that given the number of `metrics`, is not an insignificant number.

Reply:
That's fair enough, and the additional matrix will allow us to triage a failure more easily than a succint matrix. 

That said, maybe something we could do per-release could be a pytest time analysis. I've done it in cuML a few times (example; https://github.com/rapidsai/cuml/issues/3228) when pytests were slowing things down, it could be a practice that will allow us to be proactive before we accidentally slow things down when adding new pytests, but that is beyond the PR, but was wondering what do you think about that

Reply:
I think thats a great idea - as we keep adding tests, keeping on top of the total execution time will be important. We could even add something like `pytest --durations=10`  to print out the top 10 slowest tests on each CI run (though, right now that is a little noisy because the first test run initializes cublas etc - which hits causes a bit of a slowdown)